src/modules/auth/auth.module.ts

```diff
@@ -68,6 +71,16 @@ export class AuthModule implements NestModule {
         { path: '/v1/login/siwe/initiate', method: RequestMethod.POST },
         { path: '/v1/login/siwe/verify', method: RequestMethod.POST }
       )
-      .forRoutes({ path: '/*', method: RequestMethod.ALL });
+      .forRoutes({ path: '/*', method: RequestMethod.ALL })
+      .apply((_req: Request, res: Response) => {
+        if (!this.configService.get('BLOCKNUM_AUTH_ENABLED')) {
+          res
+            .status(404)
+            .send(
+              'Authentication at this endpoint is disabled. Other authentication protocols may be available'
+            );
+        }
+      })
+      .forRoutes({ path: '/v1/login', method: RequestMethod.POST });
```

Comment:
Why have the `404` logic as a middleware?
Alternatively, the logic could be in the `Controller`. Maybe middleware is better, I'm not sure. For instance, maybe middleware is faster. Just curious what your perspective is.

Reply:
maybe NestJS has some decorator where this logic could be implemented? Like guards. I am not sure...

it would be the cleanest approach.

Reply:
From my opinion `/login` hanldler is responsible for configuring session after succesful login. If endpoint should not be accessible, then this configuring and authentication is not needed

Reply:
yeah, middleware is executed very early, co nothing more is initiated if it is disabled like this

Reply:
Probably interceptor https://docs.nestjs.com/interceptors is a more appropriate option for this

Reply:
intercetpor could be a more declarative approach, but it is after Guard